### PR TITLE
feat: allow offline mode when connected to the internet

### LINKED ISSUE #890: Cannot use Policy as code from CLI
**Describe the bug**
Running datree from cli only (or helm plugin)
I canno use a local policy file, still getting an error

**To Reproduce**
Steps to reproduce the behavior from any helm chart folder :
1. `helm datree config set offline local`
2. `helm datree test . --verbose --no-record --policy-config datree/policy.yaml`

output is:
```
to use --policy-config flag you must first enable policy-as-code mode: https://hub.datree.io/policy-as-code
```
**Expected behavior**
datree run from local file policy

I'm running datree from gitlab CI so I do not want any connection to the web ui not internet connection.
I have so skip some test since some new addtition are not backward compatible with current produztion setup.
I've downloaded default policy.yaml and wnat to intruct datree to run from it keeping the offline mode on.

### PR DESCRIPTION
**Pull request by zmotu**
This pull request aims to address the use case where

> CI environments that may not be air gapped (ie. have internet access) but do not want the additional dependency on the external API introduced in their CI pipelines either for security, reproducibility or reliability reasons.

This use case was originally discussed in 
- https://github.com/datreeio/datree/issues/642
- https://github.com/datreeio/datree/issues/890

### CODE REVIEW

File: pkg/cliClient/cliClient_test.go [hunk 1]
```diff
@@ -132,11 +132,10 @@ func TestRequestEvaluationPrerunDataSuccess(t *testing.T) {
 		test_requestEvaluationPrerunData_anonymousSuccess(),
 	}
 
-	httpClientMock := mockHTTPClient{}
-
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			body, _ := json.Marshal(tt.mock.response.body)
+			httpClientMock := mockHTTPClient{}
```
Comment (zmotu) [inline pkg/cliClient/cliClient_test.go:138]: Moving this line into the loop addresses a bug in the tests where the call to Mock.AssertCalled was hitting the following edge case outlined in the Mock.AssertCalled comments. 

```
// AssertCalled asserts that the method was called.
// It can produce a false result when an argument is a pointer type and the underlying value changed after calling the mocked method.
func (m *Mock) AssertCalled(t TestingT, methodName string, arguments ...interface{}) bool {

```


The impact of this was that the tests originally expected a cliClient request to be called when "offlineMode=local" but the assertion was not failing. The test case for offline mode has now been broken into a seperate test to allow for a Mock.AssertNotCalled check to be made which is now the expected behaviour.

**Comment by adifayer** [comment]
@zmotu Thank you for submitting this PR! We will look into it and provide our review in the next couple of days🤩

**Review by royhadad — APPROVED** [review]
Looking good! 🚀🚀
I added a message to let users know whenever they're running in offline mode

image

**Comment by eyarz** [comment]
docs is also updated: https://github.com/datreeio/docs/pull/153
